server/service/src/requisition/program_settings/mod.rs

```diff
@@ -0,0 +1,322 @@
+mod map;
+mod prepare;
+
+use map::map;
+use prepare::prepare;
+
+use repository::{
+    PeriodRow, ProgramRequisitionOrderTypeRow, ProgramRequisitionSettings, ProgramSupplier,
+    RepositoryError,
+};
+
+use crate::service_provider::ServiceContext;
+
+#[derive(Debug, PartialEq)]
+pub struct OrderType {
+    pub order_type: ProgramRequisitionOrderTypeRow,
+    pub available_periods: Vec<PeriodRow>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ProgramSettings {
+    pub program_requisition_settings: ProgramRequisitionSettings,
+    pub suppliers: Vec<ProgramSupplier>,
+    pub order_types: Vec<OrderType>,
+}
+
+// TODO only return and calculate for X number of periods from now ? (back and forward)
```

Comment:
Is there an issue capturing this?

Reply:
i added a bullet point in PR description, will make sure to create an issue before merging/closing